Objects/dictobject.c

```diff
@@ -441,7 +441,7 @@ static void free_keys_object(PyDictKeysObject *keys, bool use_qsbr);
 static inline void
 dictkeys_incref(PyDictKeysObject *dk)
 {
-    if (dk->dk_refcnt == _Py_IMMORTAL_REFCNT) {
+    if (FT_ATOMIC_LOAD_SSIZE_RELAXED(dk->dk_refcnt) == _Py_IMMORTAL_REFCNT) {
```

Comment:
We could also do `dk == Py_EMPTY_KEYS` and avoid the atomics, but this seems fine too.

Reply:
I kind of like that, but am a little worried that it'll be overlooked if any other keys become immortal, so I'll just keep it as-is for now.